tests/integrated/test_suite

```diff
@@ -58,6 +63,98 @@ try:
 except RuntimeError:
     raise RuntimeError("The all-tests flag needs to be set by passing --all rather then --set-bool all_tests=True")
 
+##################################################################
+# Parallel stuff
```

Comment:
Please could you add a comment explaining the general strategy of how the tests will be run in parallel: using the `make` jobserver, calculating the cost of individual jobs, and so on